projects/js-packages/ai-client/src/hooks/use-ai-suggestions/index.ts

```diff
@@ -256,9 +256,18 @@ export default function useAiSuggestions( {
 		( event: CustomEvent ) => {
 			closeEventSource();
 
-			const fullSuggestion = removeLlamaArtifact( event?.detail );
+			let message = event?.detail;
+			let skipRequestCount = false;
+			if ( event?.detail?.message ) {
+				message = event?.detail?.message;
+				if ( event.detail.source && event.detail.source === 'chromeAI' ) {
+					skipRequestCount = true;
+				}
+			}
+
+			const fullSuggestion = removeLlamaArtifact( message );
 
-			onDone?.( fullSuggestion );
+			onDone?.( fullSuggestion, skipRequestCount );
```

Comment:
This can be simplified:

```suggestion
			onDone?.( fullSuggestion, event?.detail?.source === 'chromeAI' );
```